### DISCUSSION TITLE
Does sqlachemy provide orm based database view creating method?

### DISCUSSION THREAD

**Discussion by flyly0755** [sqlalchemy/sqlalchemy#11428] [Usage Questions]
Before I raised a clickhouse-sqlalchemy package related question about orm based database view creating method with database clickhouse,
https://github.com/xzkostyan/clickhouse-sqlalchemy/issues/261
```python
from clickhouse_sqlalchemy.ext.declarative import declarative_base
from sqlalchemy import Column
from sqlalchemy import Integer, String
from clickhouse_sqlalchemy import engines
from sqlalchemy import create_engine
from sqlalchemy.orm import sessionmaker
from sqlalchemy import select
from sqlalchemy_utils import create_view

Base = declarative_base()

class TableTest(Base):
    __tablename__ = 'tabletest'
    id = Column(Integer, primary_key=True)
    testcontent = Column(String)
    remark = Column(String)
    __table_args__ = (
        engines.MergeTree(order_by='id', primary_key='id'),
        {'comment': 'table used for testing'}
    )

viewlist = [] 
viewlist.extend([
    TableTest.id.label('id'),
    TableTest.testcontent.label('testcontent'),
    TableTest.remark.label('remark'),
])
stmt_view = select(viewlist)
cvView = create_view('viewTest', stmt_view, Base.metadata)

class ViewTest(Base):
    __tablename__ = 'viewTest'
    __table__ = cvView
    __table_args__ = {'comment': 'view used for testing'}

# clickhouse machine info
ckuser = "ckuser"
ckpwd = "ckpassword"
ckip = "ckhost"
ckport = "8123"
ckdbname = 'ckdb'

uri = f"clickhouse://{ckuser}:{ckpwd}@" \
      f"{ckip}:{ckport}/{ckdbname}"
engine = create_engine(uri, echo=False)
DBsession = sessionmaker(bind=engine)
session = DBsession()
print(session)
session.execute('SELECT 1')
# TableTest.__table__.create(engine) # success create table tableTest
# ViewTest.__table__.create(engine)  # sqlalchemy.exc.CompileError: No engine for table 'viewTest'
Base.metadata.create_all(bind=engine)  # success create both table tableTest and view viewTest
session.close()
```
What I think now is whether sqlalchemy itself provide method to create one database view one time.
For example ViewTest.\_\_view\_\_.create(engine) or something else?

**Comment by CaselIT** [comment]
Hi,

You are using `from sqlalchemy_utils import create_view` to create that view. I don't know what that function returns.
You should look at the `sqlalchemy_utils` docs

**Reply by flyly0755** [reply]
create_view is used for ORM class-mapping,  you can forget about it:)
Is there some general method to create one database view one time like atable.\_\_table\_\_.create(engine)?
Maybe can't use under  create_view situation, but I can change view orm class mapping form to use this general method.

**Reply by CaselIT** [reply]
> create_view is used for ORM class-mapping, you can forget about it:)

Well you are having problem with whatever `create_view ` is returning. If `create` does now work you should look at what `sqlalchemy_utils` is doing there.

**Comment by flyly0755** [comment]
A view depends on table(s), so usually with Base.metadata.create_all method to create  all related tables and views at one time, which is very useful. But in sometime just need to create only one view, so I want to know if there such method.
